### PR TITLE
CRDT

### PR DESCRIPTION
**Pull request by sinkingsugar**

### CODE REVIEW

File: src/core/runtime.cpp [hunk 1]
```diff
@@ -3298,3 +3306,64 @@ void incRef(ShardPtr shard) {
   }
 }
 } // namespace shards
+
+#ifdef TRACY_ENABLE
+void* operator new(std::size_t count) {
```
Comment (sinkingsugar) [inline src/core/runtime.cpp:3311]: @guusw can you check why stacktraces are not avail?

**Review by guusw — CHANGES_REQUESTED** [review]
For me it breaks at startup (during static initializers), can you maybe add a check so tracking is only enabled after main/shards init is called

tracy

File: src/core/runtime.cpp [hunk 2]
```diff
@@ -3298,3 +3306,64 @@ void incRef(ShardPtr shard) {
...
+void* operator new[](std::size_t count) {
+    void* ptr = std::malloc(count);
+    TracySecureAlloc(ptr, count);
+    return ptr;
+}
+
+void* operator new(std::size_t count, std::align_val_t alignment) {
+    std::size_t align_value = static_cast<std::size_t>(alignment);
+    std::size_t aligned_count = (count + align_value - 1) / align_value * align_value;
+    void* ptr = std::aligned_alloc(align_value, aligned_count);
```
[the remark this answers is not in the document]
Comment (sinkingsugar) [reply src/core/runtime.cpp:3326]: Github copilot had a better answer

**Comment by guusw** [comment]
Will push windows fixes to this branch

**Comment by sinkingsugar** [comment]
Merged with #664

**Pull request closed without merging**